internal/driver/fetch_test.go

```diff
@@ -210,112 +210,299 @@ func TestFetchWithBase(t *testing.T) {
+					labels: map[string][]string{},
+				},
+				{
+					values: []int64{200, 179943},
+					labels: map[string][]string{},
+				},
+				{
+					values: []int64{100, 17778444},
+					labels: map[string][]string{},
+				},
+				{
+					values: []int64{100, 75976},
+					labels: map[string][]string{},
+				},
+				{
+					values: []int64{300, 63568134},
+					labels: map[string][]string{},
+				},
+				{
+					values: []int64{-2700, -608881724},
+					labels: map[string][]string{"pprof::base": {"true"}},
+				},
+				{
+					values: []int64{-100, -23992},
+					labels: map[string][]string{"pprof::base": {"true"}},
+				},
+				{
+					values: []int64{-200, -179943},
+					labels: map[string][]string{"pprof::base": {"true"}},
+				},
+				{
+					values: []int64{-100, -17778444},
+					labels: map[string][]string{"pprof::base": {"true"}},
+				},
+				{
+					values: []int64{-100, -75976},
+					labels: map[string][]string{"pprof::base": {"true"}},
+				},
+				{
+					values: []int64{-300, -63568134},
+					labels: map[string][]string{"pprof::base": {"true"}},
+				},
+			},
+			"",
+		},
+		{
+			"diff_base and base both specified",
+			[]string{path + "cppbench.contention"},
+			[]string{path + "cppbench.contention"},
+			[]string{path + "cppbench.contention"},
+			false,
+			[]WantSample{},
+			"-base and -diff_base flags cannot both be specified",
 		},
 	}
 
 	for _, tc := range testcases {
-		t.Run(tc.desc, func(t *testing.T) {
-			pprofVariables = baseVars.makeCopy()
+		pprofVariables = baseVars.makeCopy()
 
-			base := make([]*string, len(tc.bases))
-			for i, s := range tc.bases {
-				base[i] = &s
-			}
+		base := make([]*string, len(tc.bases))
+		for i, s := range tc.bases {
+			base[i] = &s
+		}
 
-			f := testFlags{
-				stringLists: map[string][]*string{
-					"base": base,
-				},
-				bools: map[string]bool{
-					"normalize": tc.normalize,
-				},
-			}
-			f.args = tc.sources
+		diffBase := make([]*string, len(tc.diffBases))
+		for i, s := range tc.diffBases {
+			diffBase[i] = &s
+		}
 
-			o := setDefaults(&plugin.Options{
-				UI:      &proftest.TestUI{T: t, AllowRx: "Local symbolization failed|Some binary filenames not available"},
-				Flagset: f,
-			})
-			src, _, err := parseFlags(o)
+		f := testFlags{
+			stringLists: map[string][]*string{
+				"base":      base,
+				"diff_base": diffBase,
+			},
+			bools: map[string]bool{
+				"normalize": tc.normalize,
+			},
+		}
+		f.args = tc.sources
 
-			if err != nil {
-				t.Fatalf("%s: %v", tc.desc, err)
-			}
+		o := setDefaults(&plugin.Options{
+			UI:      &proftest.TestUI{T: t, AllowRx: "Local symbolization failed|Some binary filenames not available"},
+			Flagset: f,
+		})
+		src, _, err := parseFlags(o)
 
-			p, err := fetchProfiles(src, o)
-			pprofVariables = baseVars
-			if err != nil {
-				t.Fatal(err)
+		if tc.wantErrorMsg != "" {
+			if err == nil {
+				t.Errorf("%s: want error %q, got nil", tc.desc, tc.wantErrorMsg)
+				continue
```

Comment:
Use t.Run() to run the test cases? Then you can just use t.Fatalf() instead of t.Errorf() + continue. And you can avoid the boilerplate of formatting in tc.desc manually.

Reply:
I'd had some issues with this test failing on xcode6 with against go master when running in parallel.
(https://travis-ci.org/google/pprof/jobs/374039614)

Will try to use `runtime.GOMAXPROCS` to increase parallelism.

Reply:
"In parallel" - why? The only request was to use t.Run.